float64[] kp
float64[] kd
---
bool ok
string errstr
